### PR TITLE
[IMP] mail: add retry button for multiple records

### PR DESCRIPTION
**Pull request by nounoubensebia**
Add a retry button to enable the user to try and resend multiple emails at once,
purpose is to allow users to easily send back multiple emails which crashed for
a "one-shot" reason instead of asking them to resubmit them one by one.

Task-2492990

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_mail.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ def default_get(self, fields):
             self = self.with_context(dict(self._context, default_type=None))
         return super(MailMail, self).default_get(fields)
 
+    def mass_retry(self):
+        for mail in self:
+            if mail.state in ['exception', 'cancel']:
+                mail.mark_outgoing()
```
Comment (dbeguin) [inline addons/mail/models/mail_mail.py:106]: as mark_outgoing is not an 'ensure_one' thing, you can directly do :
`self.filtered(lambda mail: mail.state in ['exception', 'cancel']).mark_outgoing()`

File: addons/mail/views/mail_mail_views.xml [hunk 2]
```diff
@@ -67,6 +67,9 @@
             <field name="model">mail.mail</field>
             <field name="arch" type="xml">
                 <tree string="Emails" decoration-muted="state in ('sent', 'cancel')" decoration-info="state=='outgoing'" decoration-danger="state=='exception'">
+                    <header>
+                        <button name="mass_retry" string="Retry" type="object"/>
```
Comment (dbeguin) [inline addons/mail/views/mail_mail_views.xml:71]: shouldn't we display this only if there are email to re-send ('exception' or 'cancel') ?

Comment (nounoubensebia) [reply addons/mail/views/mail_mail_views.xml:71]: That would require a compute that verifies all emails right? if so, performances wouldn't take a hit?

Comment (dbeguin) [reply addons/mail/views/mail_mail_views.xml:71]: indeed. keep it like this and just raise an error saying that there is no email to resend in mass_retry method

Comment (nounoubensebia) [reply addons/mail/views/mail_mail_views.xml:71]: After checking with PO it appears that this is not necessary, so I think that we can keep it like this for now.

File: addons/mail/models/mail_mail.py [hunk 3]
```diff
@@ -100,6 +100,9 @@ def default_get(self, fields):
             self = self.with_context(dict(self._context, default_type=None))
         return super(MailMail, self).default_get(fields)
 
+    def mass_retry(self):
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_mail.py:103]: Maybe action_retry ?

**Review by tde-banana-odoo — COMMENTED** [review]
Riviou !

**Comment by tde-banana-odoo** [comment]
Thanks for updating !

@robodoo r+

**Pull request closed without merging**